LibVLCSharp/Shared/LibVLC.cs

```diff
@@ -623,28 +629,27 @@ static void UpdateProgress(IntPtr data, IntPtr dialogId, float position, string
         [MonoPInvokeCallback(typeof(LogCallback))]
         static void OnLogInternal(IntPtr data, LogLevel level, IntPtr ctx, IntPtr format, IntPtr args)
         {
-            if (_log == null) return;
+            if (data == IntPtr.Zero)
+                return;
+
+            var gch = GCHandle.FromIntPtr(data);
+
+            if (!gch.IsAllocated || !(gch.Target is LibVLC libvlc) || libvlc.IsDisposed)
+                return;
 
-            IntPtr buffer = IntPtr.Zero;
             try
             {
-                buffer = Marshal.AllocHGlobal(2048 + 1);
-                var count = MarshalUtils.vsprintf(buffer, format, args);
-                if (count > -1)
-                {
-                    var message = buffer.FromUtf8();
-                    GetLogContext(ctx, out var module, out var file, out var line);
+                var message = MarshalUtils.GetLogMessage(format, args);
+
+                GetLogContext(ctx, out var module, out var file, out var line);
 #if NET40
-                    Task.Factory.StartNew(() => _log?.Invoke(null, new LogEventArgs(level, message, module, file, line)));
+                Task.Factory.StartNew(() => libvlc._log?.Invoke(null, new LogEventArgs(level, message, module, file, line)));
 #else
-                    Task.Run(() => _log?.Invoke(null, new LogEventArgs(level, message, module, file, line)));
+                Task.Run(() => libvlc._log?.Invoke(null, new LogEventArgs(level, message, module, file, line)));
 #endif
-                }
             }
-            finally
+            catch
```

Comment:
Why the catch-all?

Reply:
Because we don't care if it fails, it's not critical, and easy to troubleshoot as there aren't much code here. But we still want to catch exceptions should they happen.

Reply:
I don't know if that would be easy to detect if one out of 5000 log lines gets discarded, but OK, it shouldn't happen.

I'd be in favor of having a comment inside the catch to explain why. (in fact, that's what R# suggests when you do such things)

Reply:
> I'd be in favor of having a comment inside the catch to explain why

Sure